packages/mobile/src/geth/GethNativeBridgeSigner.ts

```diff
@@ -102,12 +103,20 @@ export class GethNativeBridgeSigner implements Signer {
     return this.unlockTime + this.unlockDuration - this.unlockBufferSeconds > currentTimeInSeconds()
   }
 
-  decrypt(_ciphertext: Buffer) {
-    return Promise.reject(new Error('Decryption operation is not supported on this signer'))
+  decrypt(ciphertext: Buffer): Promise<Buffer> {
+    const ciphertextBase64 = ciphertext.toString('base64')
+    return this.geth.decrypt(this.account, ciphertextBase64).then((textBase64) => {
```

Comment:
Not 100% knowledgeable of our coding styles but I personally,

1. Prefer async await
2. Wouldn't use a return statement + curly brackets for a one liner function body

```
async decrypt(ciphertext: Buffer): Promise<Buffer> {
  const textBase64 = await this.geth.decrypt(this.account, ciphertext.toString('base64')
  return Buffer.from(textBase64, 'base64')
}
```

Seems a little cleaner to me.

Reply:
+1 to @AlexBHarley comment.